nixos/modules/system/activation/switch-to-configuration.py

```diff
@@ -0,0 +1,611 @@
+                                        socket_activated = True
+
+                            # If the unit is not socket-activated, record
+                            # that this unit needs to be started below.
+                            # We write this to a file to ensure that the
+                            # service gets restarted if we're interrupted.
+                            if not socket_activated:
+                                transitions.start_unit(unit)
+
+                            transitions.stop_unit(unit)
+
+    return transitions
+
+def path_to_unit_name(path: Path) -> UnitName:
+    # Use current version of systemctl binary before daemon is reexeced.
+    args = ["/run/current-system/sw/bin/systemd-escape",
+            "--suffix=mount", "-p", str(path)]
+    output = subprocess.check_output(args)
+    return UnitName(output.strip())
+
+# Compare the previous and new fstab to figure out which filesystems
+# need a remount or need to be unmounted.  New filesystems are mounted
+# automatically by starting local-fs.target.  FIXME: might be nicer if
+# we generated units for all mounts; then we could unify this with the
+# unit checking code above.
+def determine_mount_changes(transitions: UnitTransitions) -> None:
+    prev_fss, prev_swaps = parse_fstab(Path('/etc/fstab'))
+    new_fss, new_swaps = parse_fstab(out / Path('etc/fstab'))
+    for mountpoint in prev_fss.keys():
+        unit = path_to_unit_name(Path(mountpoint))
+        if mountpoint not in new_fss:
+            # Filesystem entry disappeared, so unmount it.
+            transitions.stop_unit(unit)
+            continue
+
+        new = new_fss[mountpoint]
+        prev = prev_fss[mountpoint]
+        if prev.fs_type != new.fs_type or \
+           prev.device != new.device:
+            # Filesystem type or device changed, so unmount and mount it.
+            transitions.stop_unit(unit)
+            transitions.start_unit(unit)
+        elif prev.options != new.options:
+            # Mount options changes, so remount it.
+            transitions.reload_unit(unit)
+
+    # Also handles swap devices.
+    for device in prev_swaps.keys():
+        if device not in new_swaps:
+            # Swap entry disappeared, so turn it off.  Can't use
+            # "systemctl stop" here because systemd has lots of alias
+            # units that prevent a stop from actually calling
+            # "swapoff".
+            print_err("stopping swap device:", device)
+            subprocess.check_call(["@utillinux@/sbin/swapoff", device])
+
+        # FIXME: update swap options (i.e. its priority).
+
+def reload_user_units() -> None:
+    args = ["@systemd@/bin/loginctl", "list-users", "--no-legend"]
+    active_users = subprocess.check_output(args).split('\n')
+    for f in active_users:
+        m = re.match(r"^\s*(?P<qid>\d+)\s+(?P<user>\S+)", f)
+        if m is None:
+            continue
+        uid = m.group(1)
+        user = m.group(2)
```

Comment:
Since you already use named captures above, you can use them here as well:

uid = m.group("uid")
user = m.group("user")